File: nattybear/level2/멀쩡한-사각형/Solution.java
class Solution {
  public long solution(int w, int h) {
    long total = (long) (w * h);
    long invalid = bresenham(w, h);
    System.out.println(total + " " + invalid);
    return total - invalid;
  }

  static long bresenham(long w, long h) {
    double line = (double)h / w;
    int x = 1;
    int y = 1;
    long n = 0;
    while (true) {
      if (x == w && y == h)
        break;
      double slope = (double)y / x;
      if (slope < line) {
        y++;
        n++;
        continue;
      }
      if (slope > line) {
        x++;
        n++;
        continue;
      }
      if (slope == line) {
        x++;
        y++;
        n++;
        continue;
      }
    }
    return n + 1;
  }
}

File: nattybear/level2/멀쩡한-사각형/solution.js
function solution(w, h) {
  const total = w * h
  const invalid = bresenham(w, h)
  return total - invalid
}

function bresenham(w, h) {
  const line = h / w
  let x = 1
  let y = 1
  let n = 0
  while (true) {
    if (x == w && y == h)
      break
    let slope = y / x
    if (slope < line) {
      y++ 
      n++
      continue
    }
    if (slope > line) {
      x++
      n++
      continue
    }
    if (slope == line) {
      x++
      y++
      n++
      continue
    }
  }
  return n + 1
}
